### PR TITLE
refactor: use struct for Config.GenerationOptions

### PR DESCRIPTION
**Pull request by danyf90**
Resolves https://github.com/tuist/tuist/issues/YYY
Request for comments document (if applies):

### Short description 📝

> Describe here the purpose of your PR.

### How to test the changes locally 🧐

> Include a set of steps for the reviewer to test the changes locally.

### Checklist ✅

- [ ] The code architecture and patterns are consistent with the rest of the codebase.
- [ ] The changes have been tested following the [documented guidelines](https://docs.tuist.io/contributors/testing-strategy/).
- [ ] The `CHANGELOG.md` has been updated to reflect the changes. In case of a breaking change, it's been flagged as such.
- [ ] In case the PR introduces changes that affect users, the documentation has been updated.
- [ ] In case the PR introduces changes that affect how the cache artifact is generated starting from the `TuistGraph.Target`, the `Constants.cacheVersion` has been updated.

### CODE REVIEW

File: Sources/TuistKit/Mappers/Factories/WorkspaceMapperFactory.swift [hunk 1]
```diff
@@ -82,17 +82,4 @@ final class WorkspaceMapperFactory: WorkspaceMapperFactorying {
 
         return mappers
     }
-
-    // MARK: - Fileprivate
-
-    fileprivate func lastUpgradeCheckOverride(config: Config) -> Version? {
```
Comment (fortmarek) [inline Sources/TuistKit/Mappers/Factories/WorkspaceMapperFactory.swift:88]: it's great we can get rid of this logic

File: projects/docs/docs/manifests/config.md [hunk 2]
```diff
@@ -67,16 +67,16 @@ Note that 'CompatibleXcodeVersions' can also be initialized with a string or arr
...
-| `.lastXcodeUpgradeCheck(Version)`           | Allows to suppress warnings in Xcode about updates to recommended settings added in or below the specified Xcode version. The warnings appear when Xcode version has been upgraded. It is recommended to set the version option to Xcode's version that is used for development of a project, for example `.lastUpgradeCheck(Version(13, 0, 0))` for Xcode 13.0.0. |
+| Case                                       | Description                                                                                                                                                                                                                                                                                                                                                        |
```
Comment (fortmarek) [inline projects/docs/docs/manifests/config.md:70]:
```suggestion
| Property                                       | Description                                                                                                                                                                                                                                                                                                                                                        |
```

These are no longer enums, so I think `Case` does not make sense.

**Review by fortmarek — APPROVED** [review]
This is such a great improvement 👏 

it simplifies both the usage and implementation.

**Review by laxmorek — APPROVED** [review]
You are a robot 🤖

File: Sources/TuistGraph/Models/Config.swift [hunk 3]
```diff
@@ -4,27 +4,11 @@ import TSCUtility
...
 /// This model allows to configure Tuist.
 public struct Config: Equatable, Hashable {
-    /// Contains options related to the project generation.
-    public enum GenerationOption: Hashable, Equatable {
-        /// Name used for the Xcode project
-        case xcodeProjectName(String)
-        case organizationName(String)
-        case developmentRegion(String)
-        case disableShowEnvironmentVarsInScriptPhases
-        case templateMacros(IDETemplateMacros)
```
Comment (danyf90) [inline Sources/TuistGraph/Models/Config.swift:14]: This was not mapped and not used anywhere, removed

**Pull request merged**